.github/workflows/preview_site_trigger.yml

```
@@ -0,0 +1,27 @@
+name: Call Build Preview
+
+on:
+  pull_request:
+    types: [opened, reopened]
+
+jobs:
+  approval_check:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Echo approval
+        run: |
+          echo "Workflow has been allowed to run for PR ${{ github.event.number }}. Setting artifacts and then continuing workflow runs"
+
+      # Save the PR number and branch name to an artifact for use in subsequent jobs
+      - name: Save PR number and Branch name
+        run: |
+          mkdir -p ./pr
+          echo "PR Number: ${{ github.event.number }}" > ./pr/PR_INFO
+          echo "Branch Name: ${{ github.head_ref }}" >> ./pr/PR_INFO
```

Small nit that I might have used different files for these like:

```suggestion
          echo "PR Number: ${{ github.event.number }}" > ./pr/number
          echo "Branch Name: ${{ github.head_ref }}" > ./pr/branch
```

Just to make extracting the info easier (no grep/sed'ing required).